### PR TITLE
Pass correct mass flow rate to calculate flow function multiplier for MS DX heating coil

### LINKED ISSUE #8580: Airflow curve input defect in AirloopHVAC:UnitarySystem when MS DX Heating coils operate at Speed = 1
Issue overview
--------------
PR #8501 fixed the issue #8335 to pass correct mass flow rate to calculate flow fraction for DX cooling coil performance curves at Speed = 1. However, the same problem occurs for a MS DX heating coil at speed=1, when a UnitarySystem is used as a parent. The input values vary with node mass flow rate as show below.
Pinture3
It is expected to get a constant input value. 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
Fixes #8580 

PR #8501 fixed the issue #8335 to pass correct mass flow rate to calculate flow fraction for DX cooling coil performance curves at Speed = 1. This issue fixes the same problem occurs for a MS DX heating coil at speed=1, when a UnitarySystem is used as a parent. 

Before
Pinture5

After
Pinture6

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [X ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [X ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [X ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/EvaporativeCoolers.cc [hunk 1]
```diff
@@ -2801,7 +2801,7 @@ namespace EvaporativeCoolers {
         auto &EvapCond(state.dataEvapCoolers->EvapCond);
 
         EvapCoolIndex = int(Par(1));
-        DryOrWetOperatingMode = OperatingMode(Par(2));
+        DryOrWetOperatingMode = OperatingMode(int(Par(2)));
```
Comment (lgu1234) [inline src/EnergyPlus/EvaporativeCoolers.cc]: A compile error occurs using Visual Studio 2017. "int" may not be needed for other MSVS versions. It depends on reviewers to determine if "int" is kept or not.

File: src/EnergyPlus/UnitarySystem.cc [hunk 2]
```diff
@@ -16104,11 +16100,7 @@ namespace UnitarySystems {
                 if (this->m_HeatingSpeedNum <= 1) {
                     this->m_HeatingSpeedRatio = 0.0;
                     this->m_HeatingCycRatio = PartLoadRatio;
-                    if (this->m_FanOpMode == DataHVACGlobals::ContFanCycCoil) {
-                        DataHVACGlobals::MSHPMassFlowRateLow = state.dataUnitarySystems->CompOnMassFlow; // #5737
-                    } else {
-                        DataHVACGlobals::MSHPMassFlowRateLow = state.dataUnitarySystems->CompOnMassFlow * PartLoadRatio; // #5518
-                    }
+                    DataHVACGlobals::MSHPMassFlowRateLow = state.dataUnitarySystems->CompOnMassFlow;
```
Comment (lgu1234) [inline src/EnergyPlus/UnitarySystem.cc]: A single statement is only needed to pass correct mass flow rate for flow function input.

File: src/EnergyPlus/UnitarySystem.cc [hunk 3]
```diff
@@ -10284,11 +10284,7 @@ namespace UnitarySystems {
         }             // No Heating/Cooling Load
 
         if (this->m_MultiSpeedHeatingCoil && (state.dataUnitarySystems->HeatingLoad && HeatSpeedNum == 1)) {
-            if (this->m_FanOpMode == DataHVACGlobals::ContFanCycCoil) {
-                DataHVACGlobals::MSHPMassFlowRateLow = state.dataUnitarySystems->CompOnMassFlow; // #5737
-            } else {
-                DataHVACGlobals::MSHPMassFlowRateLow = state.dataUnitarySystems->CompOnMassFlow * PartLoadRatio; // proportional to PLR when speed = 1,  #5518
-            }
+            DataHVACGlobals::MSHPMassFlowRateLow = state.dataUnitarySystems->CompOnMassFlow;
```
Comment (lgu1234) [inline src/EnergyPlus/UnitarySystem.cc]: A single statement is only needed to pass correct mass flow rate for flow function input.

File: src/EnergyPlus/EvaporativeCoolers.cc (hunk 1 above)
Comment (Myoldmopar) [reply src/EnergyPlus/EvaporativeCoolers.cc]: It's not great, but we need to take a wholehearted refactor on the solve-root stuff to modernize it.  C++ will offer new ways to marshal data in and out of the optimization routines that were not available in Fortran.  For today, if this works, then so be it.

File: src/EnergyPlus/UnitarySystem.cc (hunk 3 above)
Comment (Myoldmopar) [reply src/EnergyPlus/UnitarySystem.cc]: :+1:

**Review by Myoldmopar — APPROVED** [review]
No problem with these changes.

**Comment by Myoldmopar** [comment]
@lgu1234 I believe this is ready to go in.  Thanks for providing the plot showing the curve output differences.

There is a conflict, and I think at least one more global refactor branch is likely to go in soon, so I'll just hold this for right now.  I'll resolve conflicts later today and this will drop in.  @mjwitte or @rraustad do you need to see anything further on this?  It's the same as the cooling change so I think it should be straightforward.

**Comment by rraustad** [comment]
I agree, same as the cooling coil change.

**Comment by Myoldmopar** [comment]
Resolving minor conflicts now, will rebuild and test and get this merged in shortly.

**Comment by Myoldmopar** [comment]
Built just fine, 2300 some odd unit tests all work well, fix is good to go.  Thanks @lgu1234 !

**Pull request merged**